src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/WhyCommand/WhyCommandUtility/DependencyGraphFinder.cs

```diff
@@ -0,0 +1,306 @@
+        /// Returns null if the project does not have a dependency on the target package.
+        /// </returns>
+        public static Dictionary<string, List<DependencyNode>?>? GetAllDependencyGraphs(
+            LockFile assetsFile,
+            string targetPackage,
+            List<string> userInputFrameworks)
+        {
+            var dependencyGraphPerFramework = new Dictionary<string, List<DependencyNode>?>(assetsFile.Targets.Count);
+            bool doesProjectHaveDependencyOnPackage = false;
+
+            // get all top-level package and project references for the project, categorized by target framework alias
+            Dictionary<string, List<string>> topLevelReferencesByFramework = GetTopLevelPackageAndProjectReferences(assetsFile, userInputFrameworks);
+
+            if (topLevelReferencesByFramework.Count > 0)
+            {
+                foreach (var (targetFrameworkAlias, topLevelReferences) in topLevelReferencesByFramework)
+                {
+                    LockFileTarget target = assetsFile.GetTarget(targetFrameworkAlias, runtimeIdentifier: null);
+
+                    if (target != null)
+                    {
+                        // get all package libraries for the framework
+                        IList<LockFileTargetLibrary> packageLibraries = target.Libraries;
+
+                        // if the project has a dependency on the target package, get the dependency graph
+                        if (packageLibraries.Any(l => l.Name == targetPackage))
+                        {
+                            doesProjectHaveDependencyOnPackage = true;
+                            dependencyGraphPerFramework.Add(targetFrameworkAlias,
+                                                            GetDependencyGraphPerFramework(topLevelReferences, packageLibraries, targetPackage));
+                        }
+                        else
+                        {
+                            dependencyGraphPerFramework.Add(targetFrameworkAlias, null);
+                        }
+                    }
+                }
+            }
+
+            return doesProjectHaveDependencyOnPackage
+                ? dependencyGraphPerFramework
+                : null;
+        }
+
+        /// <summary>
+        /// Finds all dependency paths from the top-level packages to the target package for a given framework.
+        /// </summary>
+        /// <param name="topLevelReferences">All top-level package and project references for the framework.</param>
+        /// <param name="packageLibraries">All package libraries for the framework.</param>
+        /// <param name="targetPackage">The package we want the dependency paths for.</param>
+        /// <returns>
+        /// List of all top-level package nodes in the dependency graph.
+        /// </returns>
+        private static List<DependencyNode>? GetDependencyGraphPerFramework(
```

Comment:
naming: this doesn't make it obvious that we're getting the dependency paths for a particular package. 

It makes me think we're getting an object that represents the full graph.

Reply:
I've changed the name to GetDependencyGraph<ins>ForTarget</ins>PerFramework. Let me know if this sounds better